tools/helmify/src/kustomize_to_helm_automation.py

```diff
@@ -0,0 +1,432 @@
+
+
+def move_generated_helm_files_to_folder(dest_dir: str, source_dir: str):
+    # move crds
+    if os.path.isdir(f"{source_dir}/crds"):
+        if os.path.isdir(f"{dest_dir}/crds") == False:
+            exec_shell(f"mkdir -p {dest_dir}/crds")
+        move_crd_files(dest_dir, f"{source_dir}/crds")
+
+    # move rest of kinds
+    move_non_crd_files(f"{dest_dir}/templates", f"{source_dir}/templates")
+
+
+def move_crd_files(dest_dir: str, source_dir: str):
+    filelist = os.listdir(source_dir)
+    for file in filelist:
+        crd_dest_path = f"{dest_dir}/crds"
+        shutil.move(f"{source_dir}/{file}", f"{crd_dest_path}/{file}")
+
+    logger.info(
+        f"completed moving Custom Resource Definition yaml files into crds folder."
+    )
+
+
+def move_non_crd_files(dest_dir: str, source_dir: str):
+    obj = os.scandir(source_dir)
+    logger.info(f"moving the rest of the kinds yaml files into helm templates folder.")
+    for entry in obj:
+        if entry.is_dir() or entry.is_file():
+            # move files to template
+            # make new kind folder
+            if os.path.isdir(f"{dest_dir}/{entry.name}") == False:
+                exec_shell(f"mkdir -p {dest_dir}/{entry.name}")
+            filelist = os.listdir(f"{source_dir}/{entry.name}")
+            for file in filelist:
+                if file.endswith(".yaml"):
+                    source = f"{source_dir}/{entry.name}/{file}"
+                    dest = f"{dest_dir}/{entry.name}/{file}"
+                    shutil.move(source, dest)
+    logger.info(
+        f"completed moving the rest of the kinds yaml files into helm templates folder."
+    )
+
+
+def find_potential_failed_yaml_files(
+    helm_temp_dir: str, possible_problem_file_types: list
+):
+
+    problem_filelist = []
+    problem_filelist_output_paths = []
+
+    for file_type in possible_problem_file_types:
+        if os.path.isdir(f"{helm_temp_dir}/templates/{file_type}"):
+            logger.info(
+                f"finding potential problematic yaml files in {file_type} folder."
+            )
+            filelist = os.listdir(f"{helm_temp_dir}/templates/{file_type}")
+            for file in filelist:
+                content = load_multiple_yaml_files(
+                    file_path=f"{helm_temp_dir}/templates/{file_type}/{file}"
+                )
+                for elem in content:
+                    find_potential_failed_files_recursive_lookup(
+                        elem, problem_filelist, file
+                    )
+
+            if problem_filelist:
+                if (
+                    os.path.isdir(
+                        f"{helm_temp_dir}/potential_failed_helm_conversions/{file_type}"
+                    )
+                    == False
+                ):
+                    exec_shell(
+                        f"mkdir -p {helm_temp_dir}/potential_failed_helm_conversions/{file_type}"
+                    )
+                logger.info(
```

Comment:
Can you add a todo to print a list of all the failed components at the end.

Reply:
added in the latest commit. It will print at the end for example:
```
=================================================================
The following components have potential failed yaml files when running helm install
=================================================================
{'istio-1-14'}
please check folder in ./tools/helmify/generated_output/helm_chart_temp_output_files
```
If there are any failed components.